### PR TITLE
Clarify purpose of shader_instancing example

### PR DESCRIPTION
**Pull request by IceSentry**
# Objective

- The shader_instancing example can be misleading since it doesn't explain that bevy has built in automatic instancing.

## Solution

- Explain that bevy has built in instancing and that this example is for advanced users.
- Add a new automatic_instancing example that shows how to use the built in automatic instancing
- Rename the shader_instancing example to custom_shader_instancing to highlight that this is a more advanced implementation

### CODE REVIEW

**Review by ChristopherBiscardi — APPROVED** [review]
common question answered in the place the people asking are likely looking

**Review by tychedelia — APPROVED** [review]
Hopefully will avoid a lot of confusion in the future!

File: examples/shader/custom_shader_instancing.rs [hunk 1]
```diff
@@ -1,4 +1,11 @@
 //! A shader that renders a mesh multiple times in one draw call.
+//!
+//! Bevy will automatically batch and instance your meshes assuming you use the same
+//! `Handle<Material>` and `Handle<Mesh>` for all of your instances.
+//!
+//! This example is intended for advanced users and shows how to make custom instancing
```
Comment (JMS55) [inline examples/shader/custom_shader_instancing.rs:6]:
```suggestion
//! This example is intended for advanced users and shows how to make a custom instancing
```

File: examples/shader/custom_shader_instancing.rs [hunk 2]
```diff
@@ -1,4 +1,11 @@
...
+//! implementation using bevy's low level rendering api.
+//! It's generally recommended to rely on the built-in instancing before trying this approach.
```
Comment (JMS55) [inline examples/shader/custom_shader_instancing.rs:8]:
```suggestion
//! It's generally recommended to try the built-in instancing before going with this approach.
```

**Review by alice-i-cecile — APPROVED** [review]

**Pull request merged**